docs/DesignDocs/SGX_0-base_Enclave.md

```diff
@@ -0,0 +1,228 @@
+# OE-SGX Enclave Enforcing NULL Pointer Exception
+
+In this paper, we present a design to guarantee Page Fault on NULL point
+dereference inside an OE SGX enclave, even if the OS is malicious.
+
+## 0-Page Dereference Expectation
+
+[NullPointerException](https://docs.oracle.com/javase/7/docs/api/java/lang/NullPointerException.html)
+or
+[NullReferenceException](https://docs.microsoft.com/en-us/dotnet/api/system.nullreferenceexception?view=net-5.0)
+are used in certain high level programming languages such as C# and Java, not
+just to catch programing mistakes, but also potentially as legit programing
+techniques. For example, .NET runtime's JIT logic might generate code that does
+not always check for NULL object before accessing the object, instead, relying
+on the NullPointerException behavior when 0-page is accessed.
+
+Guaranteed Page Fault on NULL pointer dereference is also the expectation of many
+legacy applications.
+
+## SGX Enclave App Memory Layout and NULL Pointer Dereference Behavior
+
+In a typical SGX application, virtual address 0 is not part of the SGX Enclave
+linear address range of [`SECS.BASEADDR`, `SECS.BASEADDR` + `SECS.SIZE` -1]. The
+typical linear address memory layout of a SGX enclave is shown below. The blue
+pages are committed (through EADD or EAUG/EACCEPT instruction) Enclave pages.
+The red pages marked as Guard Page(s) are within the Enclave linear address
```

Comment:
Consider adding a section that points out that the existing use of Guard Pages as unallocated pages in enclave to trigger #PF will need to be modified with the upcoming EDMM work. The added paging support would need to reject (i.e. not EACCEPT) page-ins of what should be guard page ranges by a malicious host OS.

Reply:
Added "Implication for future sGX2.0 EDMM support" section in the updated PR.